### PR TITLE
For Prisma 5: refactor(introspection): remove version checker

### PR DESCRIPTION
**Pull request by Jolg42**
contributes to https://github.com/prisma/prisma/issues/19314

Removing version checker and Prisma 1 logic

TS PR: https://github.com/prisma/prisma/pull/19376

### CODE REVIEW

File: schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs [hunk 1]
```diff
@@ -55,47 +55,8 @@ pub(crate) fn is_relay_table(table: TableWalker<'_>) -> bool {
...
-pub(crate) fn is_prisma_1_point_0_join_table(table: TableWalker<'_>) -> bool {
-    table.columns().len() == 3
-        && table.indexes().len() >= 2
-        && table.columns().any(|c| c.name() == "id")
-        && common_prisma_m_to_n_relation_conditions(table)
-}
-
 /// If a relation defines a Prisma many to many relation.
-fn common_prisma_m_to_n_relation_conditions(table: TableWalker<'_>) -> bool {
+pub fn is_prisma_m_to_n_relation(table: TableWalker<'_>) -> bool {
```
Comment (tomhoule) [inline schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs:59]: Can you make this `pub(crate)` instead? So it doesn't end up in the crate's public API.

File: schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs [hunk 2]
```diff
@@ -0,0 +1,159 @@
+use either::Either;
```
Comment (tomhoule) [inline schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs:1]: I think this file is in two places now because of a rebase

Comment (tomhoule) [reply schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs:1]: The other one should be kept (src/introspection/introspection_pair/default.rs)

File: schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs (hunk 1 above)
Comment (Jolg42) [reply schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs:59]: Yes! good point

File: schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs (hunk 2 above)
Comment (Jolg42) [reply schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs:1]: Indeed, something weird happened here 🤔 
Nice catch

Comment (Jolg42) [reply schema-engine/connectors/sql-schema-connector/src/introspection_pair/default.rs]: Done

File: schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs (hunk 1 above)
Comment (Jolg42) [reply schema-engine/connectors/sql-schema-connector/src/introspection/introspection_helpers.rs]: Done ✅

**Review by tomhoule — APPROVED** [review]

**Pull request merged**